### PR TITLE
config: Add a check if custom param name is empty

### PR DESCRIPTION
**Pull request by orelmisan**
Until this commit, it was possible to receive an empty string as a valid custom parameter name.

Add a check if a custom parameter name is empty, and fail if it is.

Signed-off-by: Orel Misan <omisan@redhat.com>

### CODE REVIEW

File: kiagnose/internal/config/config_test.go [hunk 1]
```diff
@@ -177,6 +177,15 @@ func TestReadFromConfigMapShouldFail(t *testing.T) {
 			configMapData: map[string]string{types.ImageKey: imageName, types.TimeoutKey: timeoutValue, types.RolesKey: "illegal name\n"},
 			expectedError: "role name",
 		},
+		{
+			description: "when param name is empty",
+			configMapData: map[string]string{
+				types.ImageKey:           imageName,
+				types.TimeoutKey:         timeoutValue,
+				types.ParamNameKeyPrefix: "some value",
```
Comment (ormergi) [inline kiagnose/internal/config/config_test.go:185]: Shouldn't it be an empty string?

**Review by ormergi — COMMENTED** [review]
Hi Orel, thanks for addressing this.

nit about the commit message, I would rephrase it to something like "Currently the checkup parameters are not validated.."

File: kiagnose/internal/config/config_test.go (hunk 1 above)
Comment (orelmisan) [reply kiagnose/internal/config/config_test.go:185]: The value here doesn't matter - its the key that matters, and here its just the prefix.
This translates to an empty string key.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/b32e065fbdc50da26225ea7da15f04a27dd9da52..cc2983d754f093b57b0ca863a90aebc521272f88): Edited commit message.

File: kiagnose/internal/config/config_test.go (hunk 1 above)
Comment (EdDev) [reply kiagnose/internal/config/config_test.go:185]: The confusion indeed exist.
I would recommend making this explicit:
`types.ParamNameKeyPrefix + emtyParamName: ...`

**Review by EdDev — COMMENTED** [review]
Thanks! Just a small point which will make the unit test clearer.

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kiagnose/compare/cc2983d754f093b57b0ca863a90aebc521272f88..751a2426875f1174db08157aa11ca0a3ab130356): Added `emptyParamName` constant to make the test more explicit.

File: kiagnose/internal/config/config_test.go (hunk 1 above)
Comment (orelmisan) [reply kiagnose/internal/config/config_test.go]: Done.

**Review by ormergi — APPROVED** [review]
LGTM

**Review by EdDev — APPROVED** [review]
Thank you!
